### ISSUE TITLE
Interactive terminal app

### ISSUE THREAD

**Issue by CharlieKarafotias** [CharlieKarafotias/tmgr#82]

**Comment by CharlieKarafotias** [comment]
Implement using [live queries](https://surrealdb.com/blog/live-queries-in-rust). The table should auto update

**Comment by CharlieKarafotias** [comment]
Setting back to Todo as many changes occurred since first opened

**Comment by CharlieKarafotias** [comment]
Have made good progress. Left to do:
- Fix issue where need to hit `q` twice to exit. Unsure why the refactor causes this
- Reimplement the logic for Main screen and task view
- For task view, I want to be able to edit the note associated with the task. This is not MVP so it can be separate task. Think of when keybind is hit, then main part of task view adjusts to editor where note can be updated. I'd like side by side where task fields are on left and note editor view on right

**Comment by CharlieKarafotias** [comment]
Add in note functionality with [tui-textarea component](https://github.com/rhysd/tui-textarea)

**Comment by CharlieKarafotias** [comment]
Moving back to todo, need to read into [best practices of ratatui](https://ratatui.rs/concepts/) before starting this work
